zio-kafka/src/main/scala/zio/kafka/consumer/internal/Runloop.scala

```diff
@@ -224,7 +225,7 @@ private[consumer] final class Runloop(
     val buf = mutable.Map.empty[TopicPartition, Chunk[ByteArrayConsumerRecord]]
     buf ++= bufferedRecords.recs
 
-    var fulfillAction: UIO[_] = ZIO.unit
+    val fulfillActionsBuffer = ListBuffer.empty[UIO[Unit]]
```

Comment:
With this suggestion the `asInstanceOf` on line 252 can be removed.
```suggestion
    val fulfillActionsBuffer = ListBuffer.empty[UIO[Boolean]]
```